odoo/addons/base/migrations/13.0.1.3/post-migration.py

```diff
@@ -56,18 +62,59 @@ def migrate(env, version):
         WHERE binding_model_id IS NULL
             AND src_model IS NOT NULL
             AND iaw.src_model = im.model; """)
-    # Convert obsolete 'action_form_only' binding type to binding_view_types
+
+
+def fill_ir_cron_lastcall(env):
+    """Fill with current date for avoiding notifs of past recurring events."""
+    now = fields.Datetime.context_timestamp(env["ir.cron"], datetime.now())
     openupgrade.logged_query(
         env.cr,
-        """ UPDATE ir_actions
-        SET binding_view_types = 'form', binding_type = 'action'
-        WHERE binding_type = 'action_form_only' """)
-    # Set new name column from old name column if missing
+        "UPDATE ir_cron SET lastcall = %s",
+        (fields.Datetime.to_string(now.astimezone(pytz.UTC)),),
+    )
+
+
+def fill_arch_prev(env):
+    """This field is a fallback to previous architecture in case of failure,
+    so we should initially put the same as the current value.
+    """
     openupgrade.logged_query(
         env.cr,
-        sql.SQL(""" UPDATE ir_attachment
+        "UPDATE ir_ui_view SET arch_prev = arch_db")
+
+
+def fill_ir_attachment_legacy_name(env):
+    # Set new name column from old name column if missing
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE ir_attachment
         SET name = {legacy_name}
         WHERE COALESCE(name, '') = ''
-        AND COALESCE({legacy_name}, '') != '' """).format(
-            legacy_name=sql.Identifier(
-                openupgrade.get_legacy_name('name'))))
+        AND COALESCE({legacy_name}, '') != '' """.format(
```

Comment:
I'm very interested in using the composite queries that Pedro has introduced me to. Why do you replace it by ordinary string replacement? Did I not do it right first time?

Reply:
Yes, you did it right, and I can revert it, but it happens that I cannot see a reason to use the composite one instead of the ordinary one.